pkg/test/e2e/utils.go

```diff
@@ -48,36 +44,54 @@ import (
 
 type namespaceContextKey string
 
-func Main(m *testing.M, yamlFiles []string, testEnv *env.Environment) {
+type ManifestDeployDefinitions []ManifestDeployDefinition
+
+type ManifestDeployDefinition struct {
+	PreFunction  func(ctx context.Context, cfg *envconf.Config, namespace string) error
+	YamlFile     string
+	PostFunction func(ctx context.Context, cfg *envconf.Config, namespace string) error
+}
+
+func Main(m *testing.M, manifestDeployDefinitions ManifestDeployDefinitions, testEnv *env.Environment) {
 	*testEnv = env.New()
 	kindClusterName := "test"
-	namespace := "test"
+	namespace := "default"
 	dockerImage := "quay.io/netobserv/flowlogs-pipeline"
 	dockerTag := "e2e"
 
 	(*testEnv).Setup(
-		e2eCreateKindCluster(kindClusterName),
+		e2eRecreateKindCluster(kindClusterName),
 		e2eBuildAndLoadImageIntoKind(dockerImage, dockerTag, kindClusterName),
 		e2eRecreateNamespace(namespace),
-		e2eDeployEnvironmentResources(yamlFiles, namespace),
+		e2eDeployEnvironmentResources(manifestDeployDefinitions, namespace),
 	)
 
 	(*testEnv).Finish(
-		envfuncs.DeleteNamespace(namespace),
-		envfuncs.DestroyKindCluster(kindClusterName),
+		e2eDeleteKindCluster(kindClusterName),
 	)
 	os.Exit((*testEnv).Run(m))
 }
 
-func e2eCreateKindCluster(clusterName string) env.Func {
+func e2eRecreateKindCluster(clusterName string) env.Func {
 	return func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
-		fmt.Printf("====> Creating KIND cluster - %s\n", clusterName)
+		fmt.Printf("====> Recreating KIND cluster - %s\n", clusterName)
+		gexe.RunProc(fmt.Sprintf(`kind delete cluster --name %s`, clusterName))
```

Comment:
Why don't you use `envfuncs.DestroyKindCluster()`?

Reply:
Because it is not in the `ctx` yet, so we can't use this function .... in the end of the day if you look on the underline implementation ... it is doing the same thing .... just running a process ... so I do not see any value in using that